tools/compare_pages.sh

```diff
@@ -64,22 +64,54 @@ file_mod_time() {
   fi
 }
 
+# Check if cached PDF is up-to-date
+is_pdf_current() {
+  local pdf_file="$1"
+
+  if [[ ! -f "$pdf_file" ]]; then
+    return 1
+  fi
+
+  # First check: time-based check (fast)
+  local mod_time now age
+  mod_time=$(file_mod_time "$pdf_file")
+  now=$(date +%s)
+  age=$((now - mod_time))  # seconds
+
+  # If file is newer than 1 hour, consider it current
+  if [[ $age -le 3600 ]]; then
+    return 0
+  fi
+
+  # File is older than 1 hour, check commit SHA if possible
+  if command -v pdftotext >/dev/null 2>&1; then
+    # Get 7-character commit SHA from remote main branch (matching PDF format)
+    git fetch origin main 2>/dev/null
+    local pattern
+    pattern=$(git --no-pager log origin/main -1 --format="%h" 2>/dev/null)
```

Comment:
And now your assumption is that this remote is called 'origin'. It can be called 'upstream' or anything else really

What is the issue with the time-based approach?

Reply:
It's just annoying.